docs/documentation/platform/kms/hsm-integration.mdx

```diff
@@ -94,165 +94,450 @@ For organizations that work with US government agencies, FIPS compliance is almo
+
+          </Step>
+        </Steps>
+        After following these steps, your Docker setup will be ready to use HSM encryption.
+      </Tab>
+    </Tabs>
+  </Tab>
+  <Tab title="Kubernetes">
+    When you are deploying Infisical with the [Kubernetes self-hosting option](/self-hosting/deployment-options/kubernetes-helm), you can still use HSM encryption, but you need to ensure that the HSM client files are present in the container.
+
+    <Tabs>
+      <Tab title="Thales Luna Cloud HSM">
+        <Note>
+          This is only supported on helm chart version `1.4.1` and above. Please see the [Helm Chart Changelog](https://github.com/Infisical/infisical/blob/main/helm-charts/infisical-standalone-postgres/CHANGELOG.md#141-march-19-2025) for more information.
+        </Note>
+
+        <Steps>
+          <Step title="Create HSM client folder">
+            When using Kubernetes, you need to mount the path containing the HSM client files. This section covers how to configure your Infisical instance to use an HSM with Kubernetes.
+
+
+            ```bash
+            mkdir /etc/hsm-client
+            ```
+
+            After [setting up your Luna Cloud HSM client](https://thalesdocs.com/gphsm/luna/7/docs/network/Content/install/client_install/add_dpod.htm), you should have a set of files, referred to as the HSM client. You don't need all the files, but for simplicity we recommend copying all the files from the client.
+
+            A folder structure of a client folder will often look like this:
+            ```
+            partition-ca-certificate.pem
+            partition-certificate.pem
+            server-certificate.pem
+            Chrystoki.conf
+            /plugins
+              libcloud.plugin
+            /lock
+            /libs
+              /64
+                libCryptoki2.so
+            /jsp
+              LunaProvider.jar
+              /64
+                libLunaAPI.so
+            /etc
+              openssl.cnf
+            /bin
+              /64
+                ckdemo
+                lunacm
+                multitoken
+                vtl
+            ```
+
+            The most important parts of the client folder is the `Chrystoki.conf` file, and the `libs`, `plugins`, and `jsp` folders. You need to copy these files to the folder you created in the first step.
+
+            ```bash
+            cp -r /<path-to-where-your-hsm-client-is-located> /etc/hsm-client
+            ```
+          </Step>
+          <Step title="Update Chrystoki.conf">
+            The `Chrystoki.conf` file is used to configure the HSM client. You need to update the `Chrystoki.conf` file to point to the correct file paths.
+
+            In this example, we will be mounting the `/etc/hsm-client` folder to the Kubernetes deployment under a different path. The path we will use in this example is `/hsm-client`. This means `/etc/hsm-client` will be mounted to `/hsm-client` in the Kubernetes deployment.
```

Comment:
you mount volumes to containers within pods, not deployments. maybe better to day:

In this example, we will be mounting the /etc/hsm-client folder from the host to containers in our deployment's pods at the path /hsm-client. This means the contents of /etc/hsm-client on the host will be accessible at /hsm-client within the containers.